Build images successfully, but build kube and heml failed.

I can build images successfully, but when I tried to build kube or build helm, the error below showed:
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x7bafd4]

goroutine 1 [running]:
github.com/SUSE/fissile/helm.(*Encoder).writeNode(0xc4202d5880, 0x0, 0x0, 0xc4201136d0, 0x0, 0x0, 0x1)
	/root/workspace/goworkspace/src/github.com/SUSE/fissile/helm/config.go:561 +0x154
github.com/SUSE/fissile/helm.(*Encoder).Encode(0xc4202d5880, 0x0, 0x0, 0x0, 0x0)
	/root/workspace/goworkspace/src/github.com/SUSE/fissile/helm/config.go:438 +0xe0
github.com/SUSE/fissile/app.(*Fissile).generateKubeRoles(0xc42005c840, 0xc420017ac0, 0x1, 0xc420017a10, 0x7, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/root/workspace/goworkspace/src/github.com/SUSE/fissile/app/fissile.go:1078 +0x77a
github.com/SUSE/fissile/app.(*Fissile).GenerateKube(0xc42005c840, 0xc42006a5a0, 0x5e, 0x0, 0x0, 0x0, 0xc420017ac0, 0x1, 0xc420017a10, 0x7, ...)
	/root/workspace/goworkspace/src/github.com/SUSE/fissile/app/fissile.go:924 +0x391
github.com/SUSE/fissile/cmd.glob..func2(0xc8a220, 0xcb3698, 0x0, 0x0, 0x0, 0x0)
	/root/workspace/goworkspace/src/github.com/SUSE/fissile/cmd/build-helm.go:65 +0x45d
github.com/SUSE/fissile/vendor/github.com/spf13/cobra.(*Command).execute(0xc8a220, 0xcb3698, 0x0, 0x0, 0xc8a220, 0xcb3698)
	/root/workspace/goworkspace/src/github.com/SUSE/fissile/vendor/github.com/spf13/cobra/command.go:646 +0x3e8
github.com/SUSE/fissile/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc8aee0, 0x20, 0xc42005c800, 0xc420113e80)
	/root/workspace/goworkspace/src/github.com/SUSE/fissile/vendor/github.com/spf13/cobra/command.go:737 +0x2fe
github.com/SUSE/fissile/vendor/github.com/spf13/cobra.(*Command).Execute(0xc8aee0, 0x1, 0xc42005c840)
	/root/workspace/goworkspace/src/github.com/SUSE/fissile/vendor/github.com/spf13/cobra/command.go:695 +0x2b
github.com/SUSE/fissile/cmd.Execute(0xc42005c840, 0xa0dce0, 0x19, 0x4, 0x4)
	/root/workspace/goworkspace/src/github.com/SUSE/fissile/cmd/root.go:75 +0x69
main.main()
	/root/workspace/goworkspace/src/github.com/SUSE/fissile/main.go:45 +0x186

--- a/app/fissile.go
+++ b/app/fissile.go
@@ -1081,9 +1081,11 @@ func (f *Fissile) generateKubeRoles(settings kube.ExportSettings) error {
 				if err != nil {
 					return err
 				}
-				err = enc.Encode(deps)
-				if err != nil {
-					return err
+				if deps != nil {
+					err = enc.Encode(deps)
+					if err != nil {
+						return err
+					}
 				}
 
 				continue

--- a/app/fissile_test.go
+++ b/app/fissile_test.go
@@ -646,3 +646,35 @@ func TestFissileGetReleasesByName(t *testing.T) {
 		assert.NotContains(err.Error(), "test-dev", "Error message should not contain valid release name")
 	}
 }
+
+func TestFissileGenerateKubeRoles(t *testing.T) {
+	ui := termui.New(&bytes.Buffer{}, ioutil.Discard, nil)
+	workDir, err := os.Getwd()
+	assert.NoError(t, err)
+
+	// Set up the test params
+	releasePath := filepath.Join(workDir, "../test-assets/tor-boshrelease")
+	releasePathCacheDir := filepath.Join(releasePath, "bosh-cache")
+	roleManifestPath := filepath.Join(workDir, "../test-assets/role-manifests/exposed-ports-no-ports.yml")
+
+	f := NewFissileApplication(".", ui)
+	err = f.LoadReleases([]string{releasePath}, []string{""}, []string{""}, releasePathCacheDir)
+	require.NoError(t, err, "Failed to load release from %s", releasePath)
+
+	roleManifest, err := model.LoadRoleManifest(roleManifestPath, f.releases, f)
+	require.NoError(t, err, "Failed to load role manifest: %s", roleManifestPath)
+	require.NotNil(t, roleManifest)
+
+	outDir, err := ioutil.TempDir("", "fissile-test-generate-kube-roles")
+	require.NoError(t, err)
+	defer os.RemoveAll(outDir)
+
+	err = f.generateKubeRoles(kube.ExportSettings{OutputDir: outDir, RoleManifest: roleManifest})
+	assert.NoError(t, err)
+
+	for _, name := range []string{"myrole-deployment.yaml", "myrole-clustered.yaml"} {
+		path := filepath.Join(outDir, "bosh", name)
+		_, err := os.Stat(path)
+		assert.NoError(t, err, "Failed to find output %s", name)
+	}
+}

--- a/test-assets/role-manifests/exposed-ports-no-ports.yml
+++ b/test-assets/role-manifests/exposed-ports-no-ports.yml
@@ -1,0 +1,15 @@
+---
+roles:
+- name: myrole-deployment
+  jobs: []
+  run:
+    scaling:
+      min: 1
+      max: 2
+- name: myrole-clustered
+  jobs: []
+  tags: [ clustered ]
+  run:
+    scaling:
+      min: 1
+      max: 2

--- a/testhelpers/serialization_helper.go
+++ b/testhelpers/serialization_helper.go
@@ -21,12 +21,13 @@ func isYAMLSubsetInner(assert *assert.Assertions, expected, actual interface{}, 
 	expectedValue := reflect.ValueOf(expected)
 	actualValue := reflect.ValueOf(actual)
 
+	actualType := "<nil>"
+	if actualValue.IsValid() {
+		actualType = fmt.Sprintf("%s", actualValue.Type())
+	}
+
 	switch expectedValue.Kind() {
 	case reflect.Map:
-		actualType := "<nil>"
-		if actualValue.IsValid() {
-			actualType = fmt.Sprintf("%s", actualValue.Type())
-		}
 		if !assert.Equal(reflect.Map, actualValue.Kind(), "expected YAML path %s to be a %s, but is actually %s", yamlPath, expectedValue.Type(), actualType) {
 			return false
 		}
@@ -53,7 +54,7 @@ func isYAMLSubsetInner(assert *assert.Assertions, expected, actual interface{}, 
 		return success
 	case reflect.Array, reflect.Slice:
 		allowedTypes := []reflect.Kind{reflect.Array, reflect.Slice}
-		if !assert.Contains(allowedTypes, actualValue.Kind(), "expected YAML path %s to be a %s, but is actually %s", yamlPath, expectedValue.Type(), actualValue.Type()) {
+		if !assert.Contains(allowedTypes, actualValue.Kind(), "expected YAML path %s to be a %s, but is actually %s", yamlPath, expectedValue.Type(), actualType) {
 			return false
 		}
 		if !assert.Equal(expectedValue.Len(), actualValue.Len(), "expected slice at YAML path %s to have correct length", yamlPath) {

